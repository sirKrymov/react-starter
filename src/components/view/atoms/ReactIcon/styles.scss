@import '../../../../styles/mixins';

.react-icon {
  @include center;

  > svg {
    vertical-align: middle;
  }

  &--size-xs {
    font-size: 10px;
  }

  &--size-sm {
    font-size: 12px;
  }

  &--size-md {
    font-size: 16px;
  }

  &--size-lg {
    font-size: 20px;
  }

  &--size-xl {
    font-size: 24px;
  }

  &--size-xxl {
    font-size: 32px;
  }

  &--size-xxxl {
    font-size: 40px;
  }

  &--color-primary {
    color: var(--iconPrimary);
  }

  &--color-on-primary {
    color: var(--onPrimary);
  }

  &--color-secondary {
    color: var(--secondary);
  }

  &--color-thirdy {
    color: var(--thirdy);
  }

  &--color-valid {
    color: var(--valid);
  }

  &--color-error {
    color: var(--error);
  }

  &--rotate-90 {
    transform: rotate(90deg);
  }

  &--rotate-180 {
    transform: rotate(180deg);
  }

  &--rotate-270 {
    transform: rotate(270deg);
  }

  &--rotate-360 {
    transform: rotate(360deg);
  }

  &--pointer {
    cursor: pointer;
  }
}
