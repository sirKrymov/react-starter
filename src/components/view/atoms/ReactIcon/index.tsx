// @SETUP: Please don't forget to remove this component or current comment if component is used in project
import React, { ReactNode, ReactElement } from 'react';

import cn from 'classnames';

import './styles.scss';

interface IProps {
  className?: string;
  onClick?(): void;
  children: ReactNode;
  pointer?: boolean;
  rotate?: '90' | '180' | '270' | '360';
  color?: 'on-primary' | 'secondary' | 'primary' | 'thirdy' | 'valid' | 'error';
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl' | 'xxxl';
  id?: string;
}

export function ReactIcon({
  className,
  children,
  pointer,
  onClick,
  rotate,
  color = 'primary',
  size = 'lg',
  id
}: IProps): ReactElement {
  const reactIconClass = cn(
    {
      'react-icon': true,
      [`react-icon--rotate-${rotate}`]: rotate,
      [`react-icon--color-${color}`]: color,
      [`react-icon--size-${size}`]: size,
      'react-icon--pointer': pointer
    },
    className
  );

  return (
    <div className={reactIconClass} onClick={onClick} id={id}>
      {children}
    </div>
  );
}
