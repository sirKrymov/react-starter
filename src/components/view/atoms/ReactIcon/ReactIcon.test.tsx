import React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { IoMdCloseCircle } from 'react-icons/io';
import { shallowToJson } from 'enzyme-to-json';

import { ReactIcon } from '../ReactIcon';

describe('<ReactIcon> atom component', () => {
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    shallowComponent = shallow(
      <ReactIcon id="check-icon">
        <IoMdCloseCircle />
      </ReactIcon>
    );
  };

  describe('WHEN the component loads', () => {
    beforeEach(() => {
      createComponent();
    });

    it('THEN it should render without crashing', () => {
      expect(shallowComponent.html()).not.toHaveLength(0);
    });

    it('THEN it should match snapshot', () => {
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
