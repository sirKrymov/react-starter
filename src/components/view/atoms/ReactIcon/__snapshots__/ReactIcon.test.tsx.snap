// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactIcon> atom component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="react-icon react-icon--color-primary react-icon--size-lg"
  id="check-icon"
>
  <IoMdCloseCircle />
</div>
`;
